src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -851,6 +851,14 @@
           "could corrupt the graph in rare cases and should be used with "  \
           "care.")                                                          \
                                                                             \
+  product(bool, ShortRunningLongLoop, true, DIAGNOSTIC,                     \
+          "long counted loop/long range checks: don't create loop nest if " \
+          "loop runs for small enough number of iterations.")               \
+                                                                            \
+  develop(bool, StressShortRunningLongLoop, false,                          \
+          "Speculate all long counted loops are short running when bounds " \
+          "are unknown even if profile data doesn't say so.")               \
```

Comment:
Why only when bounds are unknown?

Reply:
I would like to see at least a hello-world test where you use this flag, to make sure it does not completely rot ;)